File: #24_Swap_Nodes_in_Pairs/Solution.java
class ListNode {
    int val;
    ListNode next;
    ListNode(int x) { val = x; }
}

public class Solution {
    public ListNode swapPairs(ListNode head) {
    	if (head == null || head.next == null) return head;
    	ListNode prev = null;
    	ListNode result = null;
    	ListNode cur = head;
    	while (cur != null && cur.next != null) {
    		if (result == null) result = cur.next;
    		if (prev != null) prev.next = cur.next;
    		ListNode tmp = cur.next.next;
    		cur.next.next = cur;
    		cur.next = tmp;
    		prev = cur;
    		cur = tmp;
    	}
    	return result;
    }
}

File: #24_Swap_Nodes_in_Pairs/Solution.py
class ListNode:
	def __init__(self, x):
		self.val = x
		self.next = None

class Solution:
	# @param {ListNode} head
	# @return {ListNode}
	def swapPairs(self, head):
		dummy = ListNode(-1)
		dummy.next = head
		p = dummy
		c = head
		while c and c.next:
			n = c.next
			p.next = n
			c.next = n.next
			n.next = c
			c = c.next
			p = p.next.next
		return dummy.next